Fix build with rust 1 84

gkurz opened the pull request:
This is a tentative to fix the build of podvm-payload with rust < 1.85 in the CAA repo.

gkurz committed:
attestation-agent: Downgrade SEV to 6.3.0

PR https://github.com/confidential-containers/guest-components/pull/1107
bumped SEV to 6.3.1, which requires rust 1.85.

Investigation showed that :
- SEV 6.3.0 has "Attestation Report v5" and only requires rust 1.80 or
  newer
- only functional difference between SEV 6.3.0 and 6.3.1 is a fix for
  SNP HW evidence parsing on a non-x86 platform [1] that was causing
  build failures on ARM [2]. We don't build that downstream.

Based on that, downgrade SEV to 6.3.0. This is a temporary workaround
until we have rust 1.85 in our build environment.

[1] https://github.com/virtee/sev/commit/62d701fb2e093ae8154bafc7ae47d9f97ab856bc
[2] https://github.com/confidential-containers/trustee/commit/c96dbe6bcc3d7529fdb27afb19a54a6625b29634

Signed-off-by: Greg Kurz <groug@kaod.org>

gkurz committed:
attestation-agent: Downgrade scroll to 0.12.0

PR https://github.com/confidential-containers/guest-components/pull/1010
bumped scroll from 0.12.0 to 0.13.0, which requires rust 1.85.

This PR is a mechanical bump comming from dependabot with no code
changes. Investigation on the scroll crate repo [1] was a bit
tedious becaue of missing release tags and missing entries in the
changelog file. It doesn't seem there are any major functional
change between 0.12.0 and 0.13.0.

Based on that, downgrade scroll to 0.12.0. This is a temporary workaround
until we have rust 1.85 in our build environment.

[1] https://github.com/m4b/scroll

Signed-off-by: Greg Kurz <groug@kaod.org>

littlejawa reviewed (approved):
/lgtm
To be reverted as soon as we can get a rust 1.85+ builder.
Thanks @gkurz !

gkurz committed:
Downgrade rust edition for protos

PR #1090 introduced a new crate and tagged it with edition 2024. This
breaks the build of downstream users that cannot use rust 1.85 yet.

After double-checking with the author, it appears that the crate
doesn't strictly require the 2024 edition.

Downgrade the edition to 2021 :
- this reflects better what the crate actually requires
- this allow builds with older rust versions

Signed-off-by: Greg Kurz <groug@kaod.org>
(cherry picked from commit 52313b49730bbcbef3cc9ca1d3be4820f468aae0)
Signed-off-by: Greg Kurz <groug@kaod.org>

gkurz commented:
> /lgtm To be reverted as soon as we can get a rust 1.85+ builder. Thanks @gkurz !

Note that the first commit is now a cherry-pick from upstream that can be safely kept.

littlejawa commented:
/lgtm
